File: backend-express/src/exampleData.ts
import { getRepository } from 'typeorm';
import { Category } from './entity/category';
import { User } from './entity/user';
import { Item } from './entity/item';
import bcryptjs from 'bcryptjs';

const createExampleData = async () => {
    try {
        const user = new User();

        const hash = await bcryptjs.hash('123', 10);

        user.name = 'David';
        user.email = 'david@example.com';
        user.password = hash;

        await getRepository(User).save(user);

        const item1 = new Item();
        item1.name = 'Tortillas';
        item1.price = 8.0;
        item1.imageUrl = 'https://foodhub.scene7.com/is/image/woolworthsltdprod/2006-mexican-chicken-tortillas:Desktop-1300x658';
        await getRepository(Item).save(item1);

        const item2 = new Item();
        item2.name = 'Cheeseburgers';
        item2.price = 10.0;
        item2.imageUrl = 'https://img.redro.pl/obrazy/two-tasty-cheeseburgers-with-american-cheese-700-212331665.jpg';
        await getRepository(Item).save(item2);

        const item3 = new Item();
        item3.name = 'Pizza';
        item3.price = 15.5;
        item3.imageUrl = 'https://www.mojegotowanie.pl/media/cache/default_view/uploads/media/recipe/0002/11/pizza-pepperoni.jpeg';
        await getRepository(Item).save(item3);

        const item4 = new Item();
        item4.name = 'Futomaki';
        item4.price = 50.0;
        item4.imageUrl = 'https://kaiseki.pl/wp-content/uploads/2019/07/kaiseki-menu-futomaki.jpg';
        await getRepository(Item).save(item4);

        const item5 = new Item();
        item5.name = 'Hosokami';
        item5.price = 30.0;
        item5.imageUrl = 'https://kuchnialidla.pl/img/PL/960x540/ff45002b0c10-5879e835f6fe-Kinga_Paruzel-sushi-hosomaki-futomaki-uramaki-nigiri-1250x700.jpg';
        await getRepository(Item).save(item5);

        const item6 = new Item();
        item6.name = 'Ice Cream';
        item6.price = 3.0;
        item6.imageUrl = 'https://assets.tmecosys.com/image/upload/t_web767x639/img/recipe/ras/Assets/0C9310E2-D356-4ED3-BBA0-147E7D21C67C/Derivates/E08BD85D-9D20-471A-BE8F-990ED6E2C1C3.jpg';
        await getRepository(Item).save(item6);

        const item7 = new Item();
        item7.name = 'Fruits';
        item7.price = 2.0;
        item7.imageUrl = 'https://ellalanguage.com/blog/wp-content/uploads/2021/12/fruit_czy_fruits_blogT-740x499.jpg';
        await getRepository(Item).save(item7);

        const category = new Category();
        category.name = 'Main Dish';
        category.items = [item1, item2, item3];
        await getRepository(Category).save(category);

        const category2 = new Category();
        category2.name = 'Sushi';
        category2.items = [item4, item5];
        await getRepository(Category).save(category2);

        const category3 = new Category();
        category3.name = 'Desserts';
        category3.items = [item6, item7];
        await getRepository(Category).save(category3);
    } catch (error) {
        // console.log(error);
    }
};

export default createExampleData;

File: backend/d2d/example_data.py
from sqlmodel import Session

from d2d.database import engine
from d2d.models.category import Category
from d2d.models.category import CategoryCreate
from d2d.models.item import Item
from d2d.models.user import User
from d2d.models.user import UserCreate
from d2d.models.code import CodeCreate
from d2d.models.code import Code
from d2d.routers.auth import get_password_hash


def create_example_data():
    with Session(engine) as session:
        user = UserCreate(
            name='David', email="david@example.com", password=get_password_hash("123")
        )
        db_user = User.from_orm(user)
        session.add(db_user)

        categories = ["Main Dish", "Sushi", "Desserts"]

        for category_name in categories:
            category = CategoryCreate(name=category_name)
            db_category = Category.from_orm(category)
            session.add(db_category)

        db_item = Item(name="Tortillas", price=8.0, category_id=1, image_url='https://foodhub.scene7.com/is/image/woolworthsltdprod/2006-mexican-chicken-tortillas:Desktop-1300x658')
        session.add(db_item)

        db_item = Item(name="Cheeseburgers", price=10.0, category_id=1, image_url='https://img.redro.pl/obrazy/two-tasty-cheeseburgers-with-american-cheese-700-212331665.jpg')
        session.add(db_item)

        db_item = Item(name="Pizza", price=15.5, category_id=1, image_url='https://www.mojegotowanie.pl/media/cache/default_view/uploads/media/recipe/0002/11/pizza-pepperoni.jpeg')
        session.add(db_item)

        db_item = Item(name="Futomaki", price=50.0, category_id=2, image_url='https://kaiseki.pl/wp-content/uploads/2019/07/kaiseki-menu-futomaki.jpg')
        session.add(db_item)

        db_item = Item(name="Hosokami", price=30.0, category_id=2, image_url='https://kuchnialidla.pl/img/PL/960x540/ff45002b0c10-5879e835f6fe-Kinga_Paruzel-sushi-hosomaki-futomaki-uramaki-nigiri-1250x700.jpg')
        session.add(db_item)

        db_item = Item(name="Ice Cream", price=3.0, category_id=3, image_url='https://assets.tmecosys.com/image/upload/t_web767x639/img/recipe/ras/Assets/0C9310E2-D356-4ED3-BBA0-147E7D21C67C/Derivates/E08BD85D-9D20-471A-BE8F-990ED6E2C1C3.jpg')
        session.add(db_item)

        db_item = Item(name="Fruits", price=2.0, category_id=3, image_url='https://ellalanguage.com/blog/wp-content/uploads/2021/12/fruit_czy_fruits_blogT-740x499.jpg')
        session.add(db_item)

        code = Code(code='STYCZEN15', discount=15)
        db_code = Code.from_orm(code)
        session.add(db_code)

        session.commit()
